docs/development/malware-checks.rst

```diff
@@ -0,0 +1,109 @@
+Malware Checks
+==============
+
+.. contents::
+    :local:
+
+Background
+----------
+
+In September 2019, the Python Software Foundation issued a `Request for Proposal`_ for a system to automate the detection of malicious uploads. This system was initially rolled out in February 2020 by `pull request 7400`_.
+
+Overview
+------------
+This is a high-level diagram of the automated malware check system.
+
+|VerdictLifecycle|
+
+Checks can be triggered in the following ways:
+
+* A PyPI user uploads a new File, Release or Project;
+* A schedule;
+* A PyPI administrator initiates an evaluation run.
+
+All of the above triggers call the `IMalwareCheckService`_ factory to determine how to execute the check. On production, 
+the `DatabaseMalwareCheckService`_ is returned, which runs the check and produces one or more verdicts. PyPI administrator and moderators are continuously reviewing the verdicts in the warehouse admin, making determinations about the accuracy of checks, about taking further action if needed (e.g. removing a malicious package surfaced by a verdict).
+
+Contributing
+------------
+
+Check Lifecycle
+~~~~~~~~~~~~~~~
+|CheckLifecycle|
+
+Ideas for new malware checks should first be shared by `opening an issue`_. This will initiate a discussion with PyPI administrators and among the broader Python community about the impact of the proposed check. After soliciting feedback, `open a pull request to master`_ containing the code for the new check, unit tests, and accompanying documentation. Once the code is reviewed and merged, it will automatically be deployed to production. PyPI administrators can begin evaluating the malware check by moving it into the `evaluation` state in the check admin and triggering an evaluation run.
+
+The evaluation run generates verdicts, which are viewable in the verdicts admin. After reviewing the verdicts, the administrator will make a determination and communicate it to the check developer in the initial issue. Here are the possible outcomes:
+
+* The check provides a low-quality or noisy signal (e.g. many false positives), and should be removed. At this point, the check will be moved into a `wiped_out` state, removing all verdicts generated by the evaluation run, and the code for the check will be removed in the next release.
+* The check provides a useful signal, but requires modifications. The administrator will request changes in the initial issue.
+* The check provides a useful signal, and the administrator enables it.
+
+Adding New Checks
+~~~~~~~~~~~~~~~~~
+All active checks are defined as classes in the `warehouse/malware/checks/`_ directory, and exported from ``__init__.py``. The checks in `tests/common/checks/`_ can serve as templates for developing new checks. Simply copy/paste the desired check template into `warehouse/malware/checks/`_ and edit the `dunder init file`_ to get started.
```

Comment:
Might want to note that complex checks that require more than a single file (or include data files, like the YARA check) are okay, and that subdirectories with their own dunder-init are fine.